### ISSUE TITLE
API Integration not loading

### ISSUE THREAD

**Issue by GypsyAcidQueen** [octimot/StoryToolkitAI#130]
**Update: Reinstalling python resolved the issue, sorry for taking up space on the board!**
****

**Describe the bug**
On launch, 0.19.5 will not connect to the resolve API (running Resolve 18.5.1 Build 6. When you click integrations, 'Connect to Resolve API,' it also will not connect. Sometimes throwing an error. 

**To Reproduce**
As above. 

**Expected behavior**
Connect to the Resolve API.

**Log file**

The error that appears after clicking 'connect to Resolve API' is:

Exception in Tkinter callback
Traceback (most recent call last):
  File "storytoolkitai\integrations\mots_resolve.py", line 166, in get_resolve
ModuleNotFoundError: No module named 'DaVinciResolveScript'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\ProgramData\Blackmagic Design\DaVinci Resolve\Support\Developer\Scripting\Modules\DaVinciResolveScript.py", line 7, in <module>
    import fusionscript as script_module
ModuleNotFoundError: No module named 'fusionscript'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tkinter\__init__.py", line 1921, in __call__
  File "storytoolkitai\ui\toolkit_ui.py", line 12221, in on_connect_resolve_api_press
  File "storytoolkitai\core\toolkit_ops\toolkit_ops.py", line 2850, in resolve_enable
  File "storytoolkitai\integrations\mots_resolve.py", line 68, in __init__
  File "storytoolkitai\integrations\mots_resolve.py", line 243, in get_resolve
  File "imp.py", line 172, in load_source
  File "<frozen importlib._bootstrap>", line 719, in _load
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\ProgramData\Blackmagic Design\DaVinci Resolve\Support\Developer\Scripting\Modules\DaVinciResolveScript.py", line 29, in <module>
    script_module = imp.load_dynamic("fusionscript", path + "fusionscript" + ext)
  File "imp.py", line 343, in load_dynamic
  File "<frozen importlib._bootstrap>", line 719, in _load
  File "<frozen importlib._bootstrap>", line 674, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 571, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1176, in create_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed

Please include the app.log file - you can find it by clicking on File -> Configuration Directory

2023-09-04 11:35:42,013 - StAI - DEBUG: All package requirements met. (__main__.py:59)
2023-09-04 11:35:42,035 - StAI - DEBUG: FFmpeg not found in current working directory. (storytoolkitai.py:671)
2023-09-04 11:35:42,035 - StAI - DEBUG: Looking for FFmpeg in env variable. (storytoolkitai.py:675)
2023-09-04 11:35:42,035 - StAI - DEBUG: FFMPEG_BINARY env variable is empty. Looking for FFmpeg in PATH. (storytoolkitai.py:682)
2023-09-04 11:35:42,035 - StAI - DEBUG: Checking ffmpeg binary: .\ffmpeg.EXE (storytoolkitai.py:694)
2023-09-04 11:35:42,050 - StAI - DEBUG: FFmpeg exit code: 1 (storytoolkitai.py:699)
2023-09-04 11:35:42,050 - StAI - DEBUG: FFmpeg found at .\ffmpeg.EXE (storytoolkitai.py:702)
2023-09-04 11:35:44,114 - StAI - DEBUG: Loading config file C:\Users\User\StoryToolkitAI\config.json. (storytoolkitai.py:319)
2023-09-04 11:35:44,777 - StAI - INFO: [1m[4mRunning StoryToolkitAI version 0.19.5 (standalone) (storytoolkitai.py:95)
2023-09-04 11:35:44,778 - StAI - DEBUG: Using cuda for Torch. (toolkit_ops.py:729)
2023-09-04 11:35:44,778 - StAI - DEBUG: Resolve API disabled via config. (toolkit_ops.py:192)
2023-09-04 11:35:44,843 - StAI - DEBUG: Running with TK 8.6.12 (toolkit_ui.py:1057)
2023-09-04 11:35:45,378 - StAI - INFO: Starting StoryToolkitAI GUI (toolkit_ui.py:2290)
2023-09-04 11:35:46,974 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:46,974 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:46,975 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:46,975 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:46,976 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:46,976 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:46,976 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:46,976 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:46,977 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:46,977 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:46,977 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:46,978 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:47,434 - StAI - DEBUG: MotsResolve module initialized. (mots_resolve.py:42)
2023-09-04 11:35:47,435 - StAI - DEBUG: py not found on this machine. Skipping Python version check. (mots_resolve.py:96)
2023-09-04 11:35:47,435 - StAI - DEBUG: Trying to load DaVinciResolveScript module... (mots_resolve.py:160)
2023-09-04 11:35:47,435 - StAI - DEBUG: Found DaVinci Resolve at the default location: C:\Program Files\Blackmagic Design\DaVinci Resolve\ (mots_resolve.py:225)
2023-09-04 11:35:47,435 - StAI - DEBUG: Unable to find module DaVinciResolveScript from PYTHONPATH - trying default locations next (mots_resolve.py:237)
2023-09-04 11:35:49,159 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:49,159 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:49,160 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:49,160 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:49,161 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:49,161 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:49,161 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:49,161 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:49,162 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:49,162 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:35:49,163 - StAI - DEBUG: Window type not found for window_id: None (toolkit_ui.py:1852)
2023-09-04 11:35:49,163 - StAI - DEBUG: No window id found. Not updating the menu bar. (menu.py:116)
2023-09-04 11:36:10,893 - StAI - DEBUG: MotsResolve module initialized. (mots_resolve.py:42)
2023-09-04 11:36:10,895 - StAI - DEBUG: py not found on this machine. Skipping Python version check. (mots_resolve.py:96)
2023-09-04 11:36:10,895 - StAI - DEBUG: Trying to load DaVinciResolveScript module... (mots_resolve.py:160)
2023-09-04 11:36:10,895 - StAI - DEBUG: Found DaVinci Resolve at the default location: C:\Program Files\Blackmagic Design\DaVinci Resolve\ (mots_resolve.py:225)
2023-09-04 11:36:10,895 - StAI - DEBUG: Unable to find module DaVinciResolveScript from PYTHONPATH - trying default locations next (mots_resolve.py:237)

**Screenshots**
If possible, add screenshots to help explain your problem.

**System (please complete the following information):**
 - OS: Win 10
 - 
 - 0.19.5

**Additional context**
Anything else that is relevant here.

**Comment by octimot** [comment]
Hey there!

Sorry for the late reply!

I think BM might have changed some things on the 18.5 API, so I'll look into it ASAP.

Cheers

**Comment by octimot** [comment]
I don't see this happening with Resolve 18.6+ so I'll close the issue, but feel free to re-open if it's different on your end.

Cheers!

**Comment by Jzasmack** [comment]
I am having a similar issue. I have Resolve 18.6 and StoryToolkitAI.0.19.5.WIN. When I attempt to open the Story Toolkit it runs a script or error very quickly then closes. I managed to capture a screen shot of the console and it reads: 
ModuleNotFoundError: No Module named DavinciResolveScript

There's quite a bit more text so I'll include the screen cap. Not sure what to do to fix this.
![Screenshot (54)](https://github.com/octimot/StoryToolkitAI/assets/113552735/aa971e24-3390-47f2-a265-534c77f9dfd9)

**Comment by octimot** [comment]
Please attach the app.log, or at least the lines from when the tool is started to the error. You can find it in `YOUR_USER_DIRECTORY\StoryToolkitAI\app.log`

Cheers

**Comment by Darmorque** [comment]
I'm glad I've found this thread, as it's exactly the same issua I'm facing. The standalone app works fine, but enabling Resolve integration proves troublesome. I use v19.5 on Windows 11 with clean install of Python 3.9.13 as the only python version installed. Resolve 18.6 is installed in it's default path and has scripting set to Local.

The app loads itself for a while and then closes itself. The only way to use it again is to edit config file and set ["disable_resolve_api": true,], otherwise it won't start again.

Here's the content of the log file (by the way, no idea why it says it's Windows 10 and that the python version is 3.10.11, as it's not true):

2024-07-15 22:29:08,162 - StAI - DEBUG: 
--------------
Platform: Windows 10
 Platform version: 10.0.22631
 OS: 10 10.0.22631 SP0 Multiprocessor Free  ('', '', '')  
 running Python 3.10.11.final.0
-------------- (__main__.py:15)
2024-07-15 22:29:09,218 - StAI - DEBUG: All package requirements met. (__main__.py:59)
2024-07-15 22:29:09,291 - StAI - DEBUG: FFmpeg not found in current working directory. (storytoolkitai.py:671)
2024-07-15 22:29:09,292 - StAI - DEBUG: Looking for FFmpeg in env variable. (storytoolkitai.py:675)
2024-07-15 22:29:09,292 - StAI - DEBUG: FFMPEG_BINARY env variable is empty. Looking for FFmpeg in PATH. (storytoolkitai.py:682)
2024-07-15 22:29:09,293 - StAI - DEBUG: Checking ffmpeg binary: .\ffmpeg.EXE (storytoolkitai.py:694)
2024-07-15 22:29:09,347 - StAI - DEBUG: FFmpeg exit code: 1 (storytoolkitai.py:699)
2024-07-15 22:29:09,347 - StAI - DEBUG: FFmpeg found at .\ffmpeg.EXE (storytoolkitai.py:702)
2024-07-15 22:29:15,694 - StAI - DEBUG: Loading config file C:\Users\micha\StoryToolkitAI\config.json. (storytoolkitai.py:319)
2024-07-15 22:29:16,047 - StAI - INFO: [1m[4mRunning StoryToolkitAI version 0.19.5 (standalone) (storytoolkitai.py:95)
2024-07-15 22:29:16,048 - StAI - DEBUG: Using cuda for Torch. (toolkit_ops.py:729)
2024-07-15 22:29:16,048 - StAI - DEBUG: MotsResolve module initialized. (mots_resolve.py:42)
2024-07-15 22:29:16,086 - StAI - DEBUG: Trying to load DaVinciResolveScript module... (mots_resolve.py:160)
2024-07-15 22:29:16,087 - StAI - DEBUG: Found DaVinci Resolve at the default location: C:\Program Files\Blackmagic Design\DaVinci Resolve\ (mots_resolve.py:225)
2024-07-15 22:29:16,087 - StAI - DEBUG: Unable to find module DaVinciResolveScript from PYTHONPATH - trying default locations next (mots_resolve.py:237)
2024-07-15 22:30:35,301 - StAI - DEBUG: 
--------------
Platform: Windows 10
 Platform version: 10.0.22631
 OS: 10 10.0.22631 SP0 Multiprocessor Free  ('', '', '')  
 running Python 3.10.11.final.0
-------------- (__main__.py:15)
2024-07-15 22:30:36,437 - StAI - DEBUG: All package requirements met. (__main__.py:59)
2024-07-15 22:30:36,510 - StAI - DEBUG: FFmpeg not found in current working directory. (storytoolkitai.py:671)
2024-07-15 22:30:36,510 - StAI - DEBUG: Looking for FFmpeg in env variable. (storytoolkitai.py:675)
2024-07-15 22:30:36,510 - StAI - DEBUG: FFMPEG_BINARY env variable is empty. Looking for FFmpeg in PATH. (storytoolkitai.py:682)
2024-07-15 22:30:36,511 - StAI - DEBUG: Checking ffmpeg binary: .\ffmpeg.EXE (storytoolkitai.py:694)
2024-07-15 22:30:36,561 - StAI - DEBUG: FFmpeg exit code: 1 (storytoolkitai.py:699)
2024-07-15 22:30:36,562 - StAI - DEBUG: FFmpeg found at .\ffmpeg.EXE (storytoolkitai.py:702)
2024-07-15 22:30:42,797 - StAI - DEBUG: Loading config file C:\Users\micha\StoryToolkitAI\config.json. (storytoolkitai.py:319)
2024-07-15 22:30:43,185 - StAI - INFO: [1m[4mRunning StoryToolkitAI version 0.19.5 (standalone) (storytoolkitai.py:95)
2024-07-15 22:30:43,185 - StAI - DEBUG: Using cuda for Torch. (toolkit_ops.py:729)
2024-07-15 22:30:43,185 - StAI - DEBUG: MotsResolve module initialized. (mots_resolve.py:42)
2024-07-15 22:30:43,221 - StAI - DEBUG: Trying to load DaVinciResolveScript module... (mots_resolve.py:160)
2024-07-15 22:30:43,222 - StAI - DEBUG: Found DaVinci Resolve at the default location: C:\Program Files\Blackmagic Design\DaVinci Resolve\ (mots_resolve.py:225)
2024-07-15 22:30:43,223 - StAI - DEBUG: Unable to find module DaVinciResolveScript from PYTHONPATH - trying default locations next (mots_resolve.py:237)
2024-07-15 22:35:12,473 - StAI - DEBUG: 
--------------
Platform: Windows 10
 Platform version: 10.0.22631
 OS: 10 10.0.22631 SP0 Multiprocessor Free  ('', '', '')  
 running Python 3.10.11.final.0
-------------- (__main__.py:15)
2024-07-15 22:35:13,536 - StAI - DEBUG: All package requirements met. (__main__.py:59)
2024-07-15 22:35:13,606 - StAI - DEBUG: FFmpeg not found in current working directory. (storytoolkitai.py:671)
2024-07-15 22:35:13,606 - StAI - DEBUG: Looking for FFmpeg in env variable. (storytoolkitai.py:675)
2024-07-15 22:35:13,607 - StAI - DEBUG: FFMPEG_BINARY env variable is empty. Looking for FFmpeg in PATH. (storytoolkitai.py:682)
2024-07-15 22:35:13,607 - StAI - DEBUG: Checking ffmpeg binary: .\ffmpeg.EXE (storytoolkitai.py:694)
2024-07-15 22:35:13,658 - StAI - DEBUG: FFmpeg exit code: 1 (storytoolkitai.py:699)
2024-07-15 22:35:13,658 - StAI - DEBUG: FFmpeg found at .\ffmpeg.EXE (storytoolkitai.py:702)
2024-07-15 22:35:19,651 - StAI - DEBUG: Loading config file C:\Users\micha\StoryToolkitAI\config.json. (storytoolkitai.py:319)
2024-07-15 22:35:19,985 - StAI - INFO: [1m[4mRunning StoryToolkitAI version 0.19.5 (standalone) (storytoolkitai.py:95)
2024-07-15 22:35:19,985 - StAI - DEBUG: Using cuda for Torch. (toolkit_ops.py:729)
2024-07-15 22:35:19,985 - StAI - DEBUG: Resolve API disabled via config. (toolkit_ops.py:192)
2024-07-15 22:35:19,987 - StAI - DEBUG: No items left in the queue. Try to ping the queue again later. (processing_queue.py:824)
2024-07-15 22:35:20,141 - StAI - DEBUG: Running with TK 8.6.12 (toolkit_ui.py:1057)
2024-07-15 22:35:20,502 - StAI - WARNING: A new standalone version of StoryToolkitAI is available. (toolkit_ui.py:1335)
2024-07-15 22:35:21,247 - StAI - INFO: Starting StoryToolkitAI GUI (toolkit_ui.py:2290)
2024-07-15 22:35:21,971 - StAI - DEBUG: Closing window: update_available (toolkit_ui.py:1745)
2024-07-15 22:35:38,940 - StAI - DEBUG: Closing window: preferences (toolkit_ui.py:1745)
